src/python/pants/backend/python/goals/export.py

```diff
@@ -125,6 +136,17 @@ class ExportPluginOptions:
         advanced=True,
     )
 
+    py_generated_sources = BoolOption(
```

Comment:
Should this be True by default? At least in the future via a deprecation cycle, if not now? It seems too useful not to...

Reply:
I went back and forth in my mind on whether to enable it. Including generated sources in venv in one view seemed like someone could consider it a surprise if they just expected third party deps.

On the other hand, this is a useful feature and a user can just disable it if they want. 

I'm fine with either choice for the default. Thoughts?

Reply:
I guess we should disable by default, out of caution, but document it [here](https://www.pantsbuild.org/2.20/docs/python/integrations/protobuf-and-grpc), [here](https://www.pantsbuild.org/2.20/docs/python/integrations/thrift) and the export and export-codegen goals?

Reply:
Sounds like a plan to me.

Reply:
Added some docs; let me know if more are needed.